src/main/java/picard/analysis/CollectUmiPrevalenceMetrics.java

```diff
@@ -0,0 +1,181 @@
+
+    @Override
+    protected int doWork() {
+
+        IOUtil.assertFileIsReadable(INPUT);
+
+        // get an iterator to reads that overlap the heterozygous sites
+        final Histogram<Integer> umiCount = new Histogram<>("numUmis", "duplicateSets");
+        final CountingPairedFilter countingPairedFilter = new CountingPairedFilter();
+        final CountingFilterWrapper countingAlignedFilter = new CountingFilterWrapper(new AlignedFilter(true));
+        final CountingMapQFilter countingMapQFilter = new CountingMapQFilter(MINIMUM_MQ);
+        final CountingFilterWrapper countingSecondaryOrSupplementaryFilter =
+                new CountingFilterWrapper(new SecondaryOrSupplementaryFilter());
+        final ProgressLogger progress = new ProgressLogger(log, PROGRESS_STEP_INTERVAL, "examined", "duplicate sets");
+
+        try (SamReader in = SamReaderFactory.makeDefault()
+                .referenceSequence(REFERENCE_SEQUENCE)
+                .open(INPUT)) {
+
+            IOUtil.assertFileIsWritable(OUTPUT);
+
+            final SAMRecordIterator samRecordIterator = in.iterator();
+            final List<SamRecordFilter> samFilters = CollectionUtil.makeList(
+                    countingAlignedFilter,
+                    countingMapQFilter,
+                    countingSecondaryOrSupplementaryFilter
+            );
+            if (FILTER_UNPAIRED_READS) {
+                samFilters.add(countingPairedFilter);
+            }
+
+            final FilteringSamIterator filteredSamRecordIterator = new FilteringSamIterator(samRecordIterator, new AggregateFilter(samFilters));
+            log.info("Queried BAM, getting duplicate sets.");
+
+            // get duplicate iterator from iterator above
+            final DuplicateSetIterator duplicateSets = new DuplicateSetIterator(filteredSamRecordIterator, in.getFileHeader(), false, null, log);
+
+            log.info("Starting iteration on duplicate sets");
+
+            set:
+            while (duplicateSets.hasNext()) {
+
+                final DuplicateSet set = duplicateSets.next();
+                final SAMRecord setRep = set.getRepresentative();
+
+                progress.record(setRep);
+                Set<String> barcodes = new HashSet<>();
+                for (final SAMRecord read : set.getRecords()) {
+                    if (!read.hasAttribute(BARCODE_TAG)) {
+                        log.warn("no barcode tag!");
+                        continue;
+                    }
+                    if (read.hasAttribute(BARCODE_BQ)) {
+                        final String barcodeBQ = read.getStringAttribute(BARCODE_BQ).replace(" ", "");
+                        ;
+                        final byte[] bytes = SAMUtils.fastqToPhred(barcodeBQ);
+                        final boolean badQuality = IntStream.range(0, bytes.length).map(i -> bytes[i]).anyMatch(q -> q < MINIMUM_BARCODE_BQ);
+                        if (badQuality) {
+                            log.warn("bad quality barcode");
```

Comment:
These warnings could potentially be a flood of spam.  If they were pushed into the filters it could avoid that but still capture the existence of these bad reads. 

Also, if you want the warning it would probably make sense to include additional information, maybe the bad quality value, or the read name.  Same goes for the above warning.

Reply:
👍